source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -0,0 +1,207 @@
+#include "extensions/filters/listener/tls_inspector/tls_inspector.h"
+
+#include <arpa/inet.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/network/listen_socket.h"
+#include "envoy/stats/stats.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/assert.h"
+
+//#include "common/config/well_known_names.h"
+#include "extensions/transport_sockets/well_known_names.h"
+
+#include "openssl/bytestring.h"
+#include "openssl/ssl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace TlsInspector {
+
+namespace {
+
+// This could use the same index as Ssl::ContextImpl, but allocating 1 extra
+// index seems better than adding the coupling between components.
+int sslIndex() {
```

Comment:
Since this is pretty self-contained, and we need only one value, so we can use `SSL_{set,get}_app_data` instead.